building/trusted_applications.rst

```diff
@@ -373,35 +373,39 @@ third step the binary and its signature are stitched together into the full TA.
 Offline Signing of TAs
 ======================
 
-The TA dev kit does sign an application as last step of the linking process. For
-example, the file ``ta/arch/arm/link.mk`` in the :ref:`optee_os` source tree
-contains the statement
+There are two types of TAs that can be signed offline: the in-tree TAs, which come
+with the OPTEE OS (for example the ``pkcs11`` TA) and any external TA coming from the user. Offline signing
+runs directly on `.stripped.elf` TA files and there is a slight difference as how they are
+generated. The in-tree `.stripped.elf` TA files are generated during the OPTEE OS compilation, while
+with external TAs it is usually done by the user. In both cases however, the signing process is the
+same.
 
-.. code-block:: sh
+Offline signing is done with the following sequence of steps:
 
-    $(q)$(SIGN) --key $(TA_SIGN_KEY) --uuid $(user-ta-uuid) \
-	    --in $$< --out $$@
+0. (Preparation) Generate a 2048 bit RSA key for signing in a secure, offline
+environment. Extract the public key and copy it to the ``keys`` directory in the
+:ref:`optee_os` source tree.
```

Comment:
1. Is the copy necessary? I would think specifying TA_PUBLIC_KEY` below is enough. For example `TA_PUBLIC_KEY=/path/to/public_key.pem`.
2. It would be helpful to provide an actual command which does you are describing. Personally I like to be able to give a quick look at the doc then immediately copy and adjust some examples to fit my needs.

Reply:
Done. Take a look and if necessary I'll add more info.